dotnet-desktop-guide/net/wpf/properties/framework-property-metadata.md

```diff
@@ -0,0 +1,77 @@
+
+<xref:System.Windows.FrameworkPropertyMetadata> derives directly from <xref:System.Windows.UIPropertyMetadata>, and implements the flags discussed here. Unless specifically set, `FrameworkPropertyMetadata` flags have a default value of `false`.
+
+## Reading FrameworkPropertyMetadata
+
+To retrieve metadata for a dependency property, call <xref:System.Windows.DependencyProperty.GetMetadata%2A> on the <xref:System.Windows.DependencyProperty> identifier. The `GetMetadata` call returns a `PropertyMetadata` object. If you need to query framework metadata values cast `PropertyMetadata` to <xref:System.Windows.FrameworkPropertyMetadata>.
+
+## Specifying FrameworkPropertyMetadata
+
+When you register a dependency property, you have the option to create and assign metadata to it. The metadata object that you assign can be <xref:System.Windows.PropertyMetadata> or one of its derived classes, like <xref:System.Windows.FrameworkPropertyMetadata>. Choose `FrameworkPropertyMetadata` for dependency properties that rely on WPF presentation APIs and executables for rendering and data binding. A more advanced option is to derive from `FrameworkPropertyMetadata` to create a custom metadata reporting class with more flags. Or, you might use <xref:System.Windows.UIPropertyMetadata> for non-framework properties that affect UI rendering.
+
+Although metadata options are typically set during registration of a new dependency property, you can respecify them in <xref:System.Windows.DependencyProperty.OverrideMetadata%2A> or <xref:System.Windows.DependencyProperty.AddOwner%2A> calls. When overriding metadata, always override with the same metadata type that was used during property registration.
+
+The property characteristics that are exposed by `FrameworkPropertyMetadata` are sometimes referred to as *flags*. If you're creating a `FrameworkPropertyMetadata` instance, there are two ways to populate flag values:
+
+1. Set the flags on an instance of the <xref:System.Windows.FrameworkPropertyMetadataOptions> enumeration type. `FrameworkPropertyMetadataOptions` lets you specify metadata flags in bitwise combination. Then, instantiate `FrameworkPropertyMetadata` using a constructor that has a `FrameworkPropertyMetadataOptions` parameter, and pass in your `FrameworkPropertyMetadataOptions` instance. To change metadata flags after passing `FrameworkPropertyMetadataOptions` into the <xref:System.Windows.FrameworkPropertyMetadata> constructor, change the corresponding property on the new `FrameworkPropertyMetadata` instance. For example, if you set the <xref:System.Windows.FrameworkPropertyMetadataOptions.NotDataBindable?displayProperty=nameWithType> flag, you can undo that by setting <xref:System.Windows.FrameworkPropertyMetadata.IsNotDataBindable%2A?displayProperty=nameWithType> to `false`.
```

Comment:
We should call out bitwise `OR` operation.

Reply:
Good catch! Done.